### PR TITLE
Keys API changes after the API Review

### PR DESCRIPTION
**Pull request by maririos**
Changes applied:

- `Update` methods should take `IEnumerable` instead of `IList`
- Remove `CreateKey` method that only takes `KeyCreateOptions`
- Change `DateTime` references to `DateTimeOffset`
- Change `KeyOps` variable name for `KeyOperations`
- All list operations should be `GetKeys` - ish (not contain `All`)
- `GetKeys` should return `KeyBase`, not `Key` (async and sync)
- `GetKeysVersions` should return `KeyBase`, not `Key` (async and sync)
- Remove `JsonWeb` from types like `JsonWebKeyType`, `JsonWebKeyOperations`, and `JsonWebKeyCurveName`, 
- Convert to `enum` the `JsonWeb` types

It also:
- Renames the folder of the project form `Keyvault` to `KeyVault`
- Add copyright info in each file

### CODE REVIEW

**Comment by maririos** [comment]
Testing @Azure/adp-keyvault

**Comment by maririos** [comment]
Could someone please review this PR?

**Comment by maririos** [comment]
@KrzysztofCwalina could you please take a look?

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyCreateOptions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+namespace Azure.Security.KeyVault.Keys
+{
+    public class KeyCreateOptions
+    {
+        public IList<KeyOperations> KeyOperations { get; set; }
+        public bool Hsm { get; set; }
+        public DateTimeOffset? NotBefore { get; set; }
+        public DateTimeOffset? Expires { get; set; }
+        public IDictionary<string, string> Tags { get; set; }
+
```
Comment (AlexGhiondea) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyCreateOptions.cs]: extra line

**Review by AlexGhiondea — APPROVED** [review]

**Pull request merged**